### PR TITLE
Clean up Drupal 7 Redis documentation

### PR DESCRIPTION
**Pull request by Crell**
Per https://github.com/platformsh/platformsh-docs/issues/214, the Redis docs could use work.

I tested this by running through an install myself, and correcting everything I found that didn't work as advertised.  I also decided to officially recommend PhpRedis, as the C extension should be faster according to Benchmarks Found on the Internet(tm).

### CODE REVIEW

File: reference/toolstacks/php/drupal/redis-drupal-7.md [hunk 1]
```diff
@@ -57,46 +91,50 @@ variables to match.
...
+   "sites/all/modules/contrib/redis/redis.autoload.inc"
 ]
 ```
 
 > **note**
 > Remember to tick the JSON Value box.
 
 > **note**
-> Use the actual path to your Redis module in case it is in a different location. For example: `sites/all/modules/contrib/redis`.
+> Use the actual path to your Redis module in case it is in a different location. For example: `sites/all/modules/redis`. The
```
Comment (pjcdawkins) [inline reference/toolstacks/php/drupal/redis-drupal-7.md]: Actually Drush Make defaults to `contrib` only if you ask it to. But we do ask it to in our example (https://github.com/platformsh/platformsh-example-drupal/blob/7.x/project.make#L10).

**Comment by pjcdawkins** [comment]
To me committing this kind of complex config to `settings.php` is really preferable to using the environment variables... do we _need_ to provide instructions for both?

**Comment by sruggier** [comment]
I looked at the underlying docs themselves to compare the two approaches, and it looks like they already address this question:
```
 The advantage of using environment variables is that these won't be used
 in your local build where you might not have Redis installed.
```

So I'd suggest switching the order around so that the easy way is first, but explain that if you use local builds, it may be necessary to use variables instead. Alternatively, if the PHP configuration could automatically handle both cases, then the variables wouldn't be needed, even when local builds are build used. That would let us get away with one suggested configuration method.

**Comment by Crell** [comment]
The settings.php example code, however, does nothing if the corresponding environment variables aren't defined.  It already self-checks against that.  So I think that doc is a bit misleading to being with.  Cf:

~~~php
if (!empty($_ENV['PLATFORM_RELATIONSHIPS'])) {
~~~

Arguably we could add some more conditional checks there to be sure (and we likely should anyway, looking at it), but making the PHP code run only-on-platform is quite easy.

**Comment by sruggier** [comment]
Yeah, in that case, there's no reason why someone would have to use the other way, right?

**Comment by Crell** [comment]
I suppose only if they want to; I expect most Drupalers would favor the settings.php approach, but that's just a gut feeling.  It would be different for other application stacks than Drupal, of course, but that would then be its own documentation page anyway.

**Comment by Crell** [comment]
We discussed this a bit in the public Slack channel. The consensus was to favor the settings.php variant, but keep the environment version around for completeness.  I will update the PR later today.

**Pull request merged**